airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/json/AirbyteTypeToJsonSchema.kt

```diff
@@ -10,27 +10,31 @@ import com.fasterxml.jackson.databind.node.ObjectNode
 import io.airbyte.cdk.load.data.*
 
 class AirbyteTypeToJsonSchema {
-    private fun ofType(typeName: String): ObjectNode {
-        return JsonNodeFactory.instance.objectNode().put("type", typeName)
+    private fun ofType(typeName: String, nullable: Boolean): ObjectNode {
+        val objectNode = JsonNodeFactory.instance.objectNode()
+        if (nullable) {
+            objectNode.putArray("type").add("null").add(typeName)
+        } else {
+            objectNode.put("type", typeName)
+        }
+        return objectNode
     }
 
-    fun convert(airbyteType: AirbyteType): JsonNode {
+    fun convert(airbyteType: AirbyteType, nullable: Boolean = false): JsonNode {
```

Comment:
This looks very odd, It seems to me that either the entry point should be Field, or the nullable should be derived from the `AirbyteType` with `AirbyteType` being potentially nullable.
With this currrent prototyp, it does look like the caller is responsible for figuring out if the first field is nullable when calling this.

Reply:
What you are saying makes sense to me however it is a much bigger change as it breaks the interface. What do you suggest? Should we fix this today or this small incremental fix is fine for now?

Reply:
Note that the other way around has a public that expose the nullability and a private method that does not: https://github.com/airbytehq/airbyte/blob/702b35ca7ef4ee2a5cc9ff08480873659a5c5e92/airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/json/JsonSchemaToAirbyteType.kt#L37-L39

Reply:
Updated